osquery/events/linux/auditdnetlink.cpp

```diff
@@ -473,7 +473,7 @@ bool AuditdNetlinkReader::clearAuditConfiguration() noexcept {
     }
 
     // Save the rule
-    const auto reply_size = sizeof(reply) + reply.ruledata->buflen;
```

Comment:
There is no need to change this line.  It's usually safer IMO to do sizeof() on the variable name in case someone changes the type.

Reply:
Is this general suggestion?

BTW, type of reply is audit_reply  != audit_rule_data

Reply:
No. My mistake.  I thought it was a typedef for same thing.  The whole libaudit api is really ugly, and the variables we use don't help the readability for auditd noobies like myself.  It's not in the scope of your fix, but would be nice if we cleaned this up in the future.  For example...
```
const auto ruledata_size = sizeof(*reply.ruledata) + reply.ruledata->buflen;
AuditRuleDataObject ruledata_object(ruledata_size);
```